src/vtktools.jl

```diff
@@ -187,6 +243,96 @@ function calc_node_coordinates(mesh::P4estMesh, n_visnodes)
 end
 
 
+# Calculation of the node coordinates for `TreeMesh` in 2D
+function calc_node_coordinates!(node_coordinates::AbstractArray{<:Any, 4}, nodes, mesh)
```

Comment:
If this really works for TreeMesh in 2D, I think we should stop using the `node_coordinates`-based dispatch (which is unreadable anyways)
```suggestion
# Calculation of the node coordinates for `TreeMesh` in 2D
function calc_node_coordinates!(node_coordinates, nodes, mesh::TreeMesh{2})
```